### PR TITLE
Add tspan option to DynamicSS

### PR DESCRIPTION
**Pull request by tkf**
You can limit the solver by `maxiters` but sometimes it's handy to stop it via the simulated time `tspan`.

### CODE REVIEW

File: src/solve.jl [hunk 1]
```diff
@@ -43,7 +43,12 @@ end
...
                             alg::DynamicSS,args...;kwargs...)
 
-  _prob = ODEProblem(prob.f,prob.u0,(0.0,Inf),prob.p)
-  solve(_prob,alg.alg,args...;kwargs...,
-        callback=TerminateSteadyState(alg.abstol,alg.reltol))
+  tspan = alg.tspan isa Tuple ? alg.tspan : (0.0, alg.tspan)
+  _prob = ODEProblem(prob.f,prob.u0,tspan,prob.p)
+  sol = solve(_prob,alg.alg,args...;kwargs...,
+              callback=TerminateSteadyState(alg.abstol,alg.reltol))
+  if sol.t[end] ≈ _prob.tspan[end]
```
Comment (tkf) [inline src/solve.jl]: I feel dumb to do this.  Is `=` safer?  (e.g., when `u0` happened to be a steady state)  Or we need a callback that sets a flag?  Maybe it's better to have a `:Terminated` return code which is set by `terminate!`.

**Comment by tkf** [comment]
If we go with `:Terminated` return code https://github.com/JuliaDiffEq/SteadyStateDiffEq.jl/pull/8#discussion_r226174375, I guess we can do something like the following.

In `DiffEqBase`, define:

```julia
function __terminate! end

function terminate!(integrator)
  __terminate!(integrator)
  integrator.sol = solution_new_retcode(integrator.sol, :Terminated)
end
```

In `OrdinaryDiffEq`, define `__terminate!` instead of `terminate!` (and similar in `Sundials`):

```julia
function DiffEqBase.__terminate!(integrator::ODEIntegrator)
  integrator.opts.tstops.valtree = typeof(integrator.opts.tstops.valtree)()
end
```

Also, maybe define a simple function so that it's easy to support compatibility for common cases:

```julia
iserror(sol) = !(sol.retcode in (:Success, :Terminated))
```

**Comment by ChrisRackauckas** [comment]
Honestly I think the return codes should be enums, so if we redo them we should go all the way there.

**Comment by tkf** [comment]
Yeees! enum retcode would be really nice.

File: src/solve.jl (hunk 1 above)
Comment (ChrisRackauckas) [reply src/solve.jl]: `==` is safer. The handling should make this exact.

**Comment by ChrisRackauckas** [comment]
I'll merge when tests pass. But the retcode changes of adding a terminate and changing to enums is something that should get followup when we have time.

**Comment by ChrisRackauckas** [comment]
This should get docs

**Pull request merged**